### PR TITLE
resource_control: add metrics for priority resource limiter

### LINKED ISSUE #17589: tail lantency is too high when task is limited by flow control priority limiter
## Bug Report

In one customer's production environment, we observed high tail latency when resource control is enable. But the TiKV's cpu is used about 5/8 of its quota. While we think some tasks' triggering flow-control is expected behavior, the high tail latency is not expected.

After some investigation, we found this high latency is due to our async-speed-limit mechanism:

image

https://github.com/tikv/async-speed-limit/blob/0e74acf491cfdd1fb3cce5d958cf48a82ee880bb/src/limiter.rs#L49-L57

When a tasks triggers wait, it will at least wait `self.fill` seconds and its default value is 0.1 seconds. This is too long and not acceptable in our case.

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: Close #17589

What's Changed:

```commit-message
Add some metrics for resource control priority resource limiter. Also adjust the build parameters of QuotaLimiter in resource control module to avoid triggering wait too frequently.
```

Benchmark result with different resource limiter configuration:

(NOTE: ingore the second half of the second case because there are other running process that caused this spike.)
955db431-eba9-422f-b707-db6b3b26306d

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Fix the bug that tasks will wait a long time when priority resource limiter is triggered.
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by glorv** [comment]
> In the test, the refill=100, the avg_wai_dur is about `1.5ms`? 
@nolouch The test is based on the new async-speed-limiter, and you can see the `min_wait` is set to 1ms, so the avg wait duration is about 1.5ms.

> seems use .99, .80 is more reasonable to diagnose the tail latency.

Because the difference between each avg is obvious enough so I just show the avg. The different between p99 is at similar scale.
image

File: components/tikv_util/Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ failpoints = ["fail/failpoints"]
 test-cgroup = []
 
 [dependencies]
-async-speed-limit = "0.4.0"
+async-speed-limit = { git = "https://github.com/glorv/async-speed-limit", branch = "min_wait" }
```
Comment (nolouch) [inline components/tikv_util/Cargo.toml:13]: need update it.

**Review by nolouch — COMMENTED** [review]
rest lgtm

**Review by nolouch — APPROVED** [review]

**Pull request merged**